rero_mef/api.py

```diff
@@ -121,11 +120,11 @@ def create_or_update(
     def get_record_by_pid(cls, pid):
         """Get ils record by pid value."""
         assert cls.provider
-        resolver = Resolver(pid_type=cls.provider.pid_type,
-                            object_type=cls.object_type,
-                            getter=cls.get_record)
         try:
-            persistent_identifier, record = resolver.resolve(str(pid))
+            persistent_identifier = PersistentIdentifier.get(
```

Comment:
why? It seems less general than using a resolver. Is it for performance purpose?

Reply:
To return the right record class.